server/block/explosion.go

```diff
@@ -0,0 +1,191 @@
+package block
+
+import (
+	"github.com/df-mc/dragonfly/server/block/cube"
+	"github.com/df-mc/dragonfly/server/block/cube/trace"
+	"github.com/df-mc/dragonfly/server/item"
+	"github.com/df-mc/dragonfly/server/world"
+	"github.com/df-mc/dragonfly/server/world/particle"
+	"github.com/df-mc/dragonfly/server/world/sound"
+	"github.com/go-gl/mathgl/mgl64"
+	"math"
+	"math/rand"
+	"time"
+)
+
+// ExplosionConfig is the configuration for an explosion. The world, position, size, sound, particle, and more can all
+// be configured through this configuration.
+type ExplosionConfig struct {
+	// World is the world in which the explosion should take place.
+	World *world.World
+	// Pos is the center position of the explosion.
+	Pos mgl64.Vec3
```

Comment:
I would move these to function parameters of the `Do` method so that an `ExplosionConfig` can be re-used more easily.

Reply:
Moved.